packages/typespec-ts/src/modular/buildCodeModel.ts

```diff
@@ -792,18 +792,22 @@ function emitBasicOperation(
   }
 
   let bodyParameter: any | undefined;
+  // FIXME handle the empty body case
   if (httpOperation.parameters.body === undefined) {
     bodyParameter = undefined;
   } else {
     bodyParameter = emitBodyParameter(context, httpOperation);
-    if (bodyParameter.type.type === "model" && bodyParameter.type.name === "") {
-      if (bodyParameter.type.properties.length > 0) {
-        for (const param of bodyParameter.type.properties) {
-          // const emittedParam = emitParameter(context, param.type, "Method");
-          param.implementation = "Method";
-          param.location = param.location ?? "body";
-          parameters.push(param);
-        }
+    // Flatten the body parameter if it is an anonymous model
+    if (
+      bodyParameter.type.type === "model" &&
```

Comment:
Can we use isEmptyModel here?

Reply:
No this is anonymous but with properties